src/sidebar/panel.js

```diff
@@ -31,27 +51,34 @@ function base64ToArrayBuffer(base64) {
   return bytes;
 }
 
-function arrayBufferToBase64(bytes) {
+function arrayBufferToBase64URL(bytes) {
   let binary = '';
   const len = bytes.byteLength;
   for (let i = 0; i < len; i++) {
     binary += String.fromCharCode(bytes[i]);
   }
-  return window.btoa(binary);
+  return base64URLencode(binary);
 }
 
-const ivLen = 16;
-function joinIvAndData(iv, data) {
-  const buf = new Uint8Array(iv.length + data.length);
+function buildJWE(iv, data) {
+  const ivBuf = new Uint8Array(iv.length);
   iv.forEach((byte, i) => {
-    buf[i] = byte;
+    ivBuf[i] = byte;
   });
+
+  const ciphertextBuf = new Uint8Array(data.length);
   data.forEach((byte, i) => {
-    buf[ivLen + i] = byte;
+    ciphertextBuf[i] = byte;
+  });
+
+  return JSON.stringify({
+    protected: window.btoa(JSON.stringify({ enc: 'A256GCM' })),
```

Comment:
from the spec: `  "protected", with the value BASE64URL(UTF8(JWE Protected Header))`, so I think we need to use some string `...ToBase64URL` instead of just `btoa` here?